src/pages/storage/forms/StorageVolumeFormSnapshots.tsx

```diff
@@ -1,79 +1,94 @@
 import React, { FC } from "react";
-import { Input } from "@canonical/react-components";
+import { Input, Notification } from "@canonical/react-components";
 import { FormikProps } from "formik/dist/types";
 import ConfigurationTable from "components/ConfigurationTable";
 import { getStorageConfigurationRow } from "pages/storage/forms/StorageConfigurationRow";
 import { StorageVolumeFormValues } from "pages/storage/forms/StorageVolumeForm";
 import SnapshotScheduleInput from "components/SnapshotScheduleInput";
 import { useDocs } from "context/useDocs";
+import { useProject } from "context/project";
+import { isSnapshotsDisabled } from "util/snapshots";
 
 interface Props {
   formik: FormikProps<StorageVolumeFormValues>;
 }
 
 const StorageVolumeFormSnapshots: FC<Props> = ({ formik }) => {
   const docBaseLink = useDocs();
+  const { project } = useProject();
+  const snapshotDisabled = isSnapshotsDisabled(project);
 
   return (
-    <ConfigurationTable
-      rows={[
-        getStorageConfigurationRow({
-          formik,
-          label: "Snapshot name pattern",
-          name: "snapshots_pattern",
-          defaultValue: "",
-          children: (
-            <Input
-              placeholder="Enter name pattern"
-              help={
-                <>
-                  Pongo2 template string that represents the snapshot name (used
-                  for scheduled snapshots and unnamed snapshots), see{" "}
-                  <a
-                    href={`${docBaseLink}/reference/instance_options/#instance-options-snapshots-names`}
-                    target="_blank"
-                    rel="noreferrer"
-                  >
-                    Automatic snapshot names
-                  </a>
-                </>
-              }
-              type="text"
-            />
-          ),
-        }),
+    <>
+      {snapshotDisabled && (
+        <Notification
+          severity="caution"
+          title="Before you edit snapshot settings"
+        >
+          Snapshot creation is blocked for the current project. Some of the
+          settings below may not have any effect.
```

Comment:
Can we shorten the copy or make it easier to understand? Probably by using a different title as well?

Reply:
Maybe for the title we can say 'Snapshot creation blocked for the current project'. And for the description we can say 'Snapshot scheduling settings may not work as expected'.?

Reply:
Yes, sounds better.
Maybe good to also hint how to change it -- something like "You can allow snapshot creation in the project configuration". Though, this opens another can of worms, as the user might or might not have access to alter the project configuration. We could add an "or ask your administrator to allow instance snapshots in the project configuration"? But this gets way too long. Maybe worth talking to our copy expert Sophie Felder. Or keep your above proposal as it is -- up to your choice :)

Reply:
maybe we can put a doc [link](https://documentation.ubuntu.com/lxd/en/latest/reference/projects/#project-restrictions) in the warning? Something like '[learn more](https://documentation.ubuntu.com/lxd/en/latest/reference/projects/#project-restrictions) about project restrictions'

Reply:
Yes, good idea!

Reply:
made the change.